airgun/entities/config_report.py

```diff
@@ -0,0 +1,67 @@
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep
+from airgun.navigation import navigator
+from airgun.utils import retry_navigation
+from airgun.views.config_report import ConfigReportDetailsView
+from airgun.views.config_report import ConfigReportsView
+
+
+class ConfigReportEntity(BaseEntity):
+    endpoint_path = '/config_reports'
+
+    def read(self, widget_names=None, host_name=None):
+        """Read all values for generated Config Reports"""
+        view = self.navigate_to(self, 'All')
+        return view.read(widget_names=widget_names)
+
+    def search(self, value):
+        """Search for specific Config report"""
+        view = self.navigate_to(self, 'Report Details', host_name=value)
+        return view.read()
+
+    def export(self, host_name=None):
```

Comment:
is the `hostname` supposed to be an optional parameter?

Reply:
Yeah If we are dealing with only one host for exporting its report, not all present in the table we have the option to be a little specific.